### PR TITLE
fix: Explicitly manage WorkManager.initialize

### PR DESCRIPTION
**Pull request by DaveTryon**
#### Details

As called out in #150, the current code fails to initialize on API 29 or higher. The root problem is that `WorkManager.initialize()` is not being called before the first call to `WorkManager.getInstance()`. If we simply force the call, it gets called by the default handler and fails in a different way. This PR does the following:
- Suppresses automation initialization of `WorkManager` via a manifest change
- Adds a wrapper (`WorkManagerHolder`) that ensures that we only initialize exactly once per `Context`. Since `Context` offers a `getHashCode()` method, we can safely use a `Context` as the key for a `HashMap`. It may be worth pointing out that this isn't thread-safe, but there's no indication that we need to worry about that scenario.
- Gets the `Context`-specific `WorkManager` object by calling `WorkManagerHolder.getWorkManager()` when initializing the production version of `TempFileProvider`

Confirmed that the service initializes correctly on Android API 28 and API 30

##### Motivation

Make the service work

##### Context

I had hoped to find a way to address this by controlling initialization via the manifest file, but was unable to find the correct combination of changes to make that happen.

I also considered collapsing the `WorkManagerHolder` method to just a method inside the `TempFileProvider` class, but that seemed to oppose the single-responsibility principle

#### Pull request checklist

- [x] Addresses an existing issue: #150
- [n/a] Added/updated relevant unit test(s)
- [x] Ran `./gradlew fastpass` from `AccessibilityInsightsForAndroidService`
- [x] PR title _AND_ final merge commit title both start with a semantic tag (`fix:`, `chore:`, `feat(feature-name):`, `refactor:`).

### CODE REVIEW

File: AccessibilityInsightsForAndroidService/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -58,5 +58,10 @@
             android:exported="true"
             android:permission="android.permission.SET_DEBUG_APP">
         </provider>
+        <provider
+            android:name="androidx.work.impl.WorkManagerInitializer"
+            android:authorities="${applicationId}.workmanager-init"
```
Comment (ThanyaLeif) [inline AccessibilityInsightsForAndroidService/app/src/main/AndroidManifest.xml:63]: Is the applicationId variable generated by the WorkManagerHolder?

**Review by jalkire — APPROVED** [review]
👍 validated locally

File: AccessibilityInsightsForAndroidService/app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (DaveTryon) [reply AccessibilityInsightsForAndroidService/app/src/main/AndroidManifest.xml:63]: It's defined at https://github.com/microsoft/accessibility-insights-for-android-service/blob/main/AccessibilityInsightsForAndroidService/app/build.gradle#L9:

```
 applicationId "com.microsoft.accessibilityinsightsforandroidservice"
```

The default build process then replaces the `${applicationId}` variable with the value from the gradle definition.

**Review by ThanyaLeif — APPROVED** [review]
LGTM!

**Pull request merged**